hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/DurableRingBuffer.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.durableexecutor.impl;
+
+import com.hazelcast.durableexecutor.StaleTaskIdException;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * A RingBuffer implementation to store tasks and results of those task
+ * Only a single thread (partition-operation-thread) accesses the instance
+ */
+public class DurableRingBuffer {
```

Comment:
This is not a durable ring-buffer. It's more like `TaskRingBuffer`, `CallableRingBuffer` etc..